test/integration/models/test_vpc.py

```diff
@@ -42,3 +41,69 @@ def test_update_subnet(get_client, create_vpc_with_subnet):
     subnet = get_client.load(VPCSubnet, subnet.id, vpc.id)
 
     assert subnet.label == new_label
+
+
+def test_fails_create_vpc_invalid_data(get_client):
+    try:
+        get_client.vpcs.create(
+            label="invalid_label!!",
+            region=get_region(get_client, {"VPCs"}),
+            description="test description",
+        )
+    except ApiError as e:
+        assert e.status == 400
+        assert (
+            "Label must include only ASCII letters, numbers, and dashes"
+            in str(e.json)
+        )
+
+
+def test_get_all_vpcs(get_client, create_multiple_vpcs):
+    vpc_1, vpc_2 = create_multiple_vpcs
+
+    all_vpcs = get_client.vpcs()
+
+    assert str(vpc_1) in str(all_vpcs.lists)
+    assert str(vpc_2) in str(all_vpcs.lists)
+
+
+def test_fails_update_vpc_invalid_data(create_vpc):
+    vpc = create_vpc
+
+    invalid_label = "invalid!!"
+
+    try:
+        vpc.label = invalid_label
+        vpc.save()
+    except ApiError as e:
+        assert e.status == 400
+        assert (
+            "Label must include only ASCII letters, numbers, and dashes"
```

Comment:
I'm a bit worried that asserting the exact error message may lead to test failure in the future, if the returned error message is just slightly changed on the API side. What do you think?

Reply:
Hm, yeah that could be the case I can make the assertion just on keywords